### ISSUE TITLE
Deploy default policy initiatives

### ISSUE THREAD

**Issue by brooke-hamilton** [Azure/missionlz#259]
**Benefit/Result/Outcome**  
So that a default deployment contains a set of default policies.

**Description**
MLZ will contain a default set of policy initiatives depending on which cloud is hosting the deployment. Terraform has a provider block for this: (https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/policy_set_definition)

**Acceptance Criteria**
- Primary outcome is to create report to validate compliance against NIST controls in built-in policies.
- **For the commercial cloud, deploy the NIST 800-53 and Azure Security Benchmark policy initiatives.**
- For the US Gov cloud, deploy the IL5 policy initiative. [UPDATE]  **NIST 800-53 in gov. instead of IL5**
- Policy enforcement is set to Alert. [UPDATE] Need to get Azure Activity into LA then create alerts
- **Policy deployment can be disabled by adding a flag like `--no-policy`**
- By default policies will be deployed to each subscription (including Tier 3), but only once per subscription. [UPDATE] **Scoping is set to RG level so work would need to be done per RG especially per Tier 3 deployment to be added.**
- A deployment parameter will allow deploying to an existing management group instead of at the subscription level, like `--policy-management-group <management group name>` [UPDATE] Should have scoping config in config file for change
- A deployment parameter will allow users to specify a policy initiative to deploy [UPDATE] Out of scope for now since policies have different set of parameters per cloud which can be difficult to include but should be a capability in future second 'MLZ Policy Repo' which should offer a HOW-TO enhance/create your compliance coverage.

**Comment by glennmusa** [comment]
The team discussed making policy application a `main.tf` like configuration, with individual modules for the different policy initiative types and applying them in a variable fashion.

**Comment by glennmusa** [comment]
@brooke-hamilton the acceptance criteria mentions deploying to an existing management group -- let's catch up on this when you're back from OOF 👍

**Comment by shawngib** [comment]
I think this is layered and requires some validation of the exact implementation but in general it is pretty straight forward to accomplish the acceptance criteria. Happy to lean in here as well. @brooke-hamilton can we chat about what the expected customer usage/scenario is for policies. I get the high level compliance conversation but just want to understand day 2 scenarios to help eliminate problems.

**Comment by brooke-hamilton** [comment]
@shawngib the customer scenarios vary, especially as it relates to management groups, which is why we have some variability in the acceptance criteria. For example, a customer could be deploying under an existing management group without the ability to modify the group, so they may want policies at the subscription level. More likely is that customers will be able to control the management group, but they may already have one in place or need to create it ahead of time. Either way, we want them to be able to specify the management group that will contain the policies. Another scenario is test and evaluation of MLZ, in which customers would probably prefer the policies at the subscription level if everything is being deployed to a single sub.

Let's adjust the ACs if we missed something.

**Comment by brooke-hamilton** [comment]
This was moved to done on 7/22 but the automation failed to close the issue.
